draft-fossati-tls-attestation.md

```diff
@@ -751,6 +751,62 @@ registry {{TLS-Ext-Registry}}, as follows:
 
 --- back
 
+# X.509 and Attestation usage variants
+
+The inclusion of attestation results and evidence as part of the TLS
+handshake offers the relying party information about the state of the
+workload and its cryptographic keys, but lacks the means to specify a
+stable identity to the workload in question. While the end-to-end system
+could solve this problem by including an identity as part of the
+attestation results, some use-cases require the use of traditional PKI
+mechanisms for this. It is therefore important to consider the possible
+approaches for combining regular X.509 certificates and attestation within
+one handshake.
+
+1. X.509 certificates only: no attestation metadata is carried across,
+   authentication relies on PKI alone - regular TLS with X.509 certificates
```

Comment:
> the RA verified that the key has been generated within a confidential environment

Do you mean the CA instead of RA?

Reply:
@thomas-fossati Is the policy you're thinking of machine readable? Otherwise I don't see how it can be used. And if machine readable, that would probably be a certificate extension.

Reply:
Yaron, sorry, I lost the notification for this... very belated reply below:

> Is the policy you're thinking of machine readable? Otherwise I don't see how it can be used. And if machine readable, that would probably be a certificate extension.

Yes, it's machine readable -- it'd be the policy OID in the policy cert extension.

Reply:
I get the feeling that this conversation and the one about the cert resulting from ACME device attest are actually the same thing.

Reply:
Yes, and they both belong in item #2 of the list: certificates with extra "stuff".

Reply:
> If an OID already exists with this exact semantics, let's have a reference to it.

I don't think there's one.  Each would CA define theirs.

> Otherwise, I don't think this is the right place to discuss it - item #1 on the list is plain old X.509 certificates, with no extra magic.

I'd say RFC3280's cert policies extension is well within "plain old" X.509.

Reply:
> Yes, and they both belong in item #2 of the list: certificates with extra "stuff".

it looks like we could make some sub-categories in the "extra stuff" bucket:

* custom extension
  * added to the EE cert (RA-TLS)
  * added to the CA certs chain (DICE)
* explicit using existing signalling, e.g. via policy OID in the EE cert (Carl's RedHat example)
* implicit signal, e.g., via issuer name

Reply:
Ok, I've expanded that list, don't know if the extra nesting will become fugly when rendered.

Let me know what you think - maybe I should make the title of that subsection about using X.509 extensions more verbose 🤔